builtin/providers/aws/resource_aws_inspector_assessment_template.go

```diff
@@ -0,0 +1,135 @@
+				Optional: true,
+				Computed: true,
+			},
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"duration": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"rules_package_arns": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsInspectorAssessmentTemplateCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).inspectorconn
+
+	rules := []*string{}
+	if attr := d.Get("rules_package_arns").(*schema.Set); attr.Len() > 0 {
+		rules = expandStringList(attr.List())
+	}
+
+	targetArn := d.Get("target_arn").(string)
+	templateName := d.Get("name").(string)
+	duration := int64(d.Get("duration").(int))
+
+	resp, err := conn.CreateAssessmentTemplate(&inspector.CreateAssessmentTemplateInput{
+		AssessmentTargetArn:    aws.String(targetArn),
+		AssessmentTemplateName: aws.String(templateName),
+		DurationInSeconds:      aws.Int64(duration),
+		RulesPackageArns:       rules,
+	})
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] Inspector Assessment Template %s created", *resp.AssessmentTemplateArn)
+
+	d.Set("arn", resp.AssessmentTemplateArn)
+
+	d.SetId(*resp.AssessmentTemplateArn)
+
+	return resourceAwsInspectorAssessmentTemplateRead(d, meta)
+}
+
+func resourceAwsInspectorAssessmentTemplateRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).inspectorconn
+
+	resp, err := conn.DescribeAssessmentTemplates(&inspector.DescribeAssessmentTemplatesInput{
+		AssessmentTemplateArns: []*string{
+			aws.String(d.Get("arn").(string)),
+		},
+	},
+	)
+	if err != nil {
+		if inspectorerr, ok := err.(awserr.Error); ok && inspectorerr.Code() == "InvalidInputException" {
+			return nil
+		} else {
+			log.Printf("[ERROR] Error finding Inspector Assessment Template: %s", err)
+			return err
+		}
+	}
+
+	d.Set("name", *resp.AssessmentTemplates[0].Name)
+
+	return nil
+}
+
+func resourceAwsInspectorAssessmentTemplateUpdate(d *schema.ResourceData, meta interface{}) error {
+	readTemplateErr := resourceAwsInspectorAssessmentTemplateRead(d, meta)
+	if readTemplateErr != nil {
+		return readTemplateErr
+	}
+
+	deleteTemplateErr := resourceAwsInspectorAssessmentTemplateDelete(d, meta)
+	if deleteTemplateErr != nil {
+		return deleteTemplateErr
+	}
+
+	return resourceAwsInspectorAssessmentTemplateCreate(d, meta)
+}
+
+func resourceAwsInspectorAssessmentTemplateDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).inspectorconn
+
+	_, err := conn.DeleteAssessmentTemplate(&inspector.DeleteAssessmentTemplateInput{
+		AssessmentTemplateArn: aws.String(d.Get("arn").(string)),
+	})
+	if err != nil {
+		if inspectorerr, ok := err.(awserr.Error); ok && inspectorerr.Code() == "AssessmentRunInProgressException" {
```

Comment:
Assessment runs can lasts for hours, so I would say no.  AWS recommend the minimum also be an hour, so in this instance, the error message should be enough to inform the user they need to stop the run, or wait for completion before this change.